engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/UnresolvedConstructor.java

```diff
@@ -120,13 +134,30 @@ private static <T> T[] join(T[] arr1, T[] arr2) {
     return ret;
   }
 
+  /** */
   public abstract static class ConstructNode extends Node {
```

Comment:
Shouldn't it be an `ExpressionNode`?

Reply:
I don't think this really matters. It is an internal node used to handle "data flowing thru the system". It doesn't really have a location. Making it `ExpressionNode` would probably not help. Anyway thanks for the observation.

Reply:
At the end the problem was in _not having an `ExpressionNode`_ somewhere! I had to create one [ApplicationNode](https://github.com/enso-org/enso/pull/9452/files#diff-01b243d2ba10f38e4c32a0b81a1d5960115be26e24cbe17841022e36f6a424ceR175) in `buildApplication` method to make sure real constructor value gets to `IdExecutionInstrument`.